packages/satori-js/client.ts

```diff
@@ -0,0 +1,180 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SatoriApi, ApiSession, ApiAuthenticateRequest, ApiEventRequest, ApiAuthenticateLogoutRequest, ApiAuthenticateRefreshRequest, ApiIdentifyRequest, ApiUpdatePropertiesRequest } from "./api.gen";
+
+import { Session } from "./session";
+
+const DEFAULT_HOST = "127.0.0.1";
+const DEFAULT_PORT = "7450";
+const DEFAULT_SERVER_KEY = "defaultkey";
+const DEFAULT_TIMEOUT_MS = 7000;
+const DEFAULT_EXPIRED_TIMESPAN_MS = 5 * 60 * 1000;
+
+/** A client for Satori server. */
+export class Client {
+
+  /** The expired timespan used to check session lifetime. */
+  public expiredTimespanMs = DEFAULT_EXPIRED_TIMESPAN_MS;
+
+  /** The low level API client for Nakama server. */
+  private readonly apiClient: SatoriApi;
+
+  constructor(
+      readonly serverkey = DEFAULT_SERVER_KEY,
+      readonly host = DEFAULT_HOST,
+      readonly port = DEFAULT_PORT,
+      readonly useSSL = false,
+      readonly timeout = DEFAULT_TIMEOUT_MS,
+      readonly autoRefreshSession = true) {
+    const scheme = (useSSL) ? "https://" : "http://";
+    const basePath = `${scheme}${host}:${port}`;
+
+    this.apiClient = new SatoriApi(serverkey, basePath, timeout);
+  }
+
+  /** Authenticate a user with an ID against the server. */
+  async authenticate(id: string) {
+
+    const request : ApiAuthenticateRequest = {
+      "id": id,
+    };
+
+    return this.apiClient.satoriAuthenticate(this.serverkey, "", request).then((apiSession : ApiSession) => {
+      return Promise.resolve(new Session(apiSession.token || "", apiSession.refresh_token || ""));
+    });
+  }
+
+  /** Refresh a user's session using a refresh token retrieved from a previous authentication request. */
+  async sessionRefresh(session: Session) {
+
+    const request : ApiAuthenticateRefreshRequest = {
+      "refresh_token": session.refresh_token,
+    };
+
+    return this.apiClient.satoriAuthenticateRefresh(this.serverkey, "", request).then((apiSession : ApiSession) => {
+      return Promise.resolve(new Session(apiSession.token || "", apiSession.refresh_token || ""));
+    });
+  }
+
+  /** Log out a session, invalidate a refresh token, or log out all sessions/refresh tokens for a user. */
+  async logout(session: Session) {
+
+    const request : ApiAuthenticateLogoutRequest = {
+      "token": session.token,
+      "refresh_token": session.refresh_token
+    };
+
+    return this.apiClient.satoriAuthenticateLogout(session.token, request).then((response) => {
+      return Promise.resolve(response !== undefined);
+    });
+  }
+
+  /** Publish an event for this session. */
+  async event(session: Session, name: string, value?: string, metadata: any = {}) {
```

Comment:
I think let's add the version that accepts multiple events as well.

Reply:
Added 👍